src/bentoml/_internal/cloud/deployment.py

```diff
@@ -1,73 +1,222 @@
 from __future__ import annotations
 
+import json
+import typing as t
+
+import attr
+
+from ..tag import Tag
+from ..utils import bentoml_cattr
 from .config import get_rest_api_client
 from .config import default_context_name
 from .config import default_kube_namespace
+from .schemas import DeploymentMode
+from .schemas import schema_to_json
 from .schemas import DeploymentSchema
-from .schemas import schema_from_json
 from .schemas import DeploymentListSchema
+from .schemas import DeploymentTargetType
 from .schemas import CreateDeploymentSchema
+from .schemas import DeploymentTargetConfig
+from .schemas import DeploymentTargetHPAConf
+from .schemas import DeploymentTargetCanaryRule
+from .schemas import DeploymentTargetRunnerConfig
 from ...exceptions import BentoMLException
 
 
-class deployment:
+@attr.define
+class Resource:
+    @classmethod
+    def for_hpa_conf(cls, **kwargs) -> DeploymentTargetHPAConf:
+        return bentoml_cattr.structure(kwargs, DeploymentTargetHPAConf)
+
+    @classmethod
+    def for_runner(cls, **kwargs) -> DeploymentTargetRunnerConfig:
+        exclusive_api_server_key = {
+            v for v in kwargs if v not in attr.fields_dict(DeploymentTargetRunnerConfig)
+        }
+        return bentoml_cattr.structure(
+            {k: v for k, v in kwargs.items() if k not in exclusive_api_server_key},
+            DeploymentTargetRunnerConfig,
+        )
+
+    @classmethod
+    def for_api_server(cls, **kwargs) -> DeploymentTargetConfig:
+        return bentoml_cattr.structure(kwargs, DeploymentTargetConfig)
+
+
+class Deployment:
```

Comment:
It seems like to me that this is a  mixin class, meaning all of the function in jere can be either staticmethod or classmethod